lib/wrappers/transform-node.js

```diff
@@ -71,7 +75,17 @@ module.exports = class TransformNodeWrapper extends NodeWrapper {
         fs.mkdirSync(this.outputPath);
       }
 
-      resolve(this.callbackObject.build());
+      if (this.nodeInfo.trackInputChanges === true) {
+        let obj = {};
+        this.inputNodeWrappers.forEach((wrap, idx) => {
+          let revision = this.inputRevisions.get(wrap);
+          obj[idx] = revision ? revision.changed : true;
+        });
+
+        resolve(this.callbackObject.build({ changedNodes: obj }));
```

Comment:
I wonder if this object should expose an interface like:
```js
class BuildContext
{
  changedNodes: [],
  hasInputNodeChanged(idx) {
    return !!this.changedNodes[idx];
  }
}
```
This way we can alter the data structure and interface of this object as we see fit.
We could potentially pass other info in here, like the build/rebuild number, if it's the initial or a rebuild, etc.

Just thinking out loud rn

Reply:
Im not sure how much this abstraction gives us. Im not against a method like `hasInputNodeChanged` but it feels weird to have that as what is passed into the build method.